sys/shell/commands/sc_nanocoap_vfs.c

```diff
@@ -121,7 +121,11 @@ static int _nanocoap_get_handler(int argc, char **argv)
         }
         dst = buffer;
     } else {
+        char *filename = strrchr(url, '/');
         dst = argv[2];
+        if (_is_dir(dst) && filename) {
+            strncat(dst, filename + 1, sizeof(buffer) - strlen(dst));
```

Comment:
I feel like `dst` must have been copied to `buffer`and then `dst` must point to `buffer`, for this to be safe.
And according to the manpage of `strncat(dst, src, n)`, the size of `dst` must be at least:
`strlen(dst) + n + 1`.
So for our case, if `dst` points to `buffer`, it should be:
`strncat(dst, filename + 1, sizeof(buffer) - strlen(dst) - 1);`

Reply:
Ah yea the `strncat` is API notoriously bad…

Reply:
One thing fixed. But what about the fact that right now, we append something to `argv[2]`. It certainly has nothing to do with `sizeof(buffer)`.

Reply:
Oops, good catch!